### PR TITLE
Implement `.` string constant for the current label scope

### LINKED ISSUE #775: [Feature request] A predeclared EQUS symbol for "the current global label"
For instance, `_GLOBAL` or `_SCOPE`. This would be undefined if there is no current globally scoped label.

This would be useful to avoid repeating names. I've had multiple use cases for code like this:

`Foobar:: mac Foobar`

In this case `mac` might be defining more labels that use `Foobar` as a prefix or suffix. I like to avoid having `mac` do `\1::` to declare the main label itself, since that makes `Foobar`'s definition harder to find.

Or, `mac` could be mentioning `Foobar` in some error message. This is useful for macros that don't declare any code/data/labels, but just check assertions about what's already there, like data table lengths or verifying that certain related symbols are defined.

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #775

Note that this restores the `EQUS` callbacks which were removed in 13904dc536ec652111aab1ec99e3a203369a8e9c.

### CODE REVIEW

**Review by ISSOtm — DISMISSED** [review]

File: src/asm/lexer.cpp
Comment (Rangi42) [reply src/asm/lexer.cpp:1175]: I would make that change if/when I make a PR for `..`.

Comment (ISSOtm) [reply src/asm/lexer.cpp:1175]: But doing this now would keep more error messages (and behaviour) consistent with when we do add `..` et al.

Comment (Rangi42) [reply src/asm/lexer.cpp:1175]: I'll go ahead and add `..` to this PR before merge then.

Comment (Rangi42) [reply src/asm/lexer.cpp:1175]: Actually no I won't, because we don't even track a "local scope" yet, so it would be a nontrivial change on top of this new feature. I'll just go ahead and change this then.

Comment (Rangi42) [reply src/asm/lexer.cpp]: Waiting for this to merge before implementing #1503.

**Review by ISSOtm — APPROVED** [review]
Thank you!

**Pull request merged**
